src/common/components/signUpModal/signUpModal.component.spec.js

```diff
@@ -2,132 +2,268 @@ import angular from 'angular'
+      $ctrl.sessionService.session = {
+        first_name: undefined,
+        last_name: undefined,
+        email: undefined,
+      }
+      $ctrl.$onInit()
+      expect($ctrl.donorDetails.name['given-name']).toEqual(undefined)
+      expect($ctrl.donorDetails.name['family-name']).toEqual(undefined)
+      expect($ctrl.donorDetails.email).toEqual(undefined)
+    })
 
-        describe('403 Bad Request', () => {
-          it('sets 403 error', () => {
-            deferred.reject({ status: 403 })
-            $rootScope.$digest()
+    it('Loads user cart details', () => {
+      $ctrl.isInsideAnotherModal = false
+      jest.spyOn($ctrl.cartService, 'getTotalQuantity').mockImplementation(() => Observable.from([5]))
+ 
+      $ctrl.$onInit()
 
-            expect($ctrl.hasError).toEqual(true)
-            expect($ctrl.signUpErrors).toEqual(expect.objectContaining({ 403: true }))
-          })
-        })
+      expect($ctrl.cartCount).toEqual(5)
+    })
 
-        describe('409 Bad Request', () => {
-          it('sets 409 error', () => {
-            deferred.reject({ status: 409 })
-            $rootScope.$digest()
+    it('should default to 0 items in cart if error fetching cart details', () => {
+      $ctrl.isInsideAnotherModal = false
+      jest.spyOn($ctrl.cartService, 'getTotalQuantity').mockReturnValue(Observable.throw({status: 404}))
+      $ctrl.$onInit()
+      expect($ctrl.cartCount).toEqual(0)
+    })
+  })
 
-            expect($ctrl.hasError).toEqual(true)
-            expect($ctrl.signUpErrors).toEqual(expect.objectContaining({ 409: true }))
-          })
-        })
+  describe('loadDonorDetails()', () => {
+    it('should set loadingDonorDetails to false', () => {
+      jest.spyOn($ctrl.orderService, 'getDonorDetails').mockReturnValue(Observable.throw({status: 404}))
+      $ctrl.loadingDonorDetails = true
+      $ctrl.loadDonorDetails()
+      expect($ctrl.loadingDonorDetails).toEqual(false)
+    })
+  });
 
-        describe('500 Bad Request', () => {
-          it('sets generic error', () => {
-            deferred.reject({ status: 500 })
-            $rootScope.$digest()
+  describe('submitDetails()', () => {
+    it('should return as form is not valid', () => {
+      $ctrl.submitDetails()
+      expect($ctrl.signUpForm.$setSubmitted).toHaveBeenCalled()
+      expect($ctrl.submitting).toEqual(false)   
+    })
 
-            expect($ctrl.hasError).toEqual(true)
-            expect($ctrl.signUpErrors).toEqual(expect.objectContaining({ generic: true }))
-          })
-        })
+    it('should not call createAccount', async () => {
+      jest.spyOn($ctrl.sessionService, 'createAccount')
+
+      jest.spyOn($ctrl.orderService, 'getDonorDetails').mockImplementation(() => Observable.from(
+        [signUpFormData]
+      ))
+      $ctrl.$onInit()
+      $ctrl.signUpForm.$valid = false;
+      $ctrl.submitDetails(signUpFormData.email, signUpFormData.name['given-name'], signUpFormData.name['family-name']).then(() =>{
```

Comment:
Add space after `=>`. Not sure why the lint checks are not capturing all of these code style issues.